Sources/AblyChat/DefaultMessages.swift

```diff
@@ -0,0 +1,237 @@
+import Ably
+
+// Typealias for the timeserial used to sync message subscriptions with. This is a string representation of a timestamp.
+private typealias TimeserialString = String
+
+// Wraps the MessageSubscription with the timeserial of when the subscription was attached or resumed.
+private struct MessageSubscriptionWrapper {
+    let subscription: MessageSubscription
+    var timeserial: TimeserialString
+}
+
+// MainActor is required to resolve a Swift 6.0 error "Incorrect actor executor assumption", whilst also allowing for mutations on listeners within the concrete class.
+@MainActor
+public final class DefaultMessages: Messages, EmitsDiscontinuities {
+    private let roomID: String
+    public let channel: RealtimeChannelProtocol
+    private let chatAPI: ChatAPI
+    private let clientID: String
+    private var subscriptionPoints: [UUID: MessageSubscriptionWrapper] = [:]
+
+    public nonisolated init(chatAPI: ChatAPI, roomID: String, clientID: String) {
+        self.chatAPI = chatAPI
+        self.roomID = roomID
+        self.clientID = clientID
+
+        // (CHA-M1) Chat messages for a Room are sent on a corresponding realtime channel <roomId>::$chat::$chatMessages. For example, if your room id is my-room then the messages channel will be my-room::$chat::$chatMessages.
+        let messagesChannelName = "\(roomID)::$chat::$chatMessages"
+        channel = chatAPI.getChannel(messagesChannelName)
+
+        // Implicitly handles channel events and therefore listners within this class. Alternative is to explicitly call something like `DefaultMessages.start()` which makes the SDK more cumbersome to interact with. This class is useless without kicking off this flow so I think leaving it here is suitable.
+        Task {
```

Comment:
it's "implicitly asynchronous" (I assume due to handleChannelEvents having `Task`s... I've added a comment and made the init async instead (to avoid needing to handle the cancellation)

Reply:
> it's "implicitly asynchronous"

what does that mean? is that a term of art or your own description?

Reply:
Ah, I see, saw the comment 👍